File: Data Structures & Algorithms/house-robber/submission-5.java
class Solution {
    public int rob(int[] nums) {
        int length = nums.length;
        if(length == 1 ) return nums[0];
        if(length == 2 ) return Math.max(nums[0],nums[1]);
        if(length == 3 ) return Math.max(nums[0]+nums[2],nums[1]);
        // int []dp = new int[length];
        int third = nums[0];
        int second = Math.max(nums[0],nums[1]);
        int first = Math.max(nums[0]+nums[2],nums[1]);
        int temp = 0;
        for(int i = 3; i< length; i++){
           temp = first; 
            first = Math.max(nums[i]+second,nums[i]+third);
            third = second;
            second = temp;
        } 
        return Math.max(first,second);
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        const length = nums.length;
        if(length <2){
            return length;
        }
        let set = new Set(nums);
        let arr = [];
        set.forEach((v)=>arr.push(v));
         console.log("arr1 : ",arr);
        
       arr.sort((a,b)=>a-b);
       console.log("arr2 : ",arr);
        let i=1;
        let max = 0;
        let current = 1;
        while(i<length){
             if(arr[i]===arr[i-1]+1){
                current++;
                if(i==length-1){
                    return Math.max(max,current);
                }
             }else{
                max = Math.max(max,current);
                current=1;
             }
             i++;
        }
        return max;

    }
}
